provisioning/Samples/device/SymmetricKeyWithOperationalCertificateSample/ProvisioningDeviceClientSample.cs

```diff
@@ -0,0 +1,182 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Provisioning.Client.Transport;
+using Microsoft.Azure.Devices.Shared;
+using Microsoft.Extensions.Logging;
+using System.Diagnostics;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+
+namespace Microsoft.Azure.Devices.Provisioning.Client.Samples
+{
+    /// <summary>
+    /// Demonstrates how to register a device with the device provisioning service using a symmetric key, and then
+    /// use the registration information to authenticate to IoT Hub.
+    /// </summary>
+    internal class ProvisioningDeviceClientSample
+    {
+        private readonly Parameters _parameters;
+        private readonly ILogger _logger;
+        private readonly DirectoryInfo s_dpsClientCertificateFolder;
+
+        public ProvisioningDeviceClientSample(Parameters parameters, ILogger logger)
+        {
+            _parameters = parameters;
+            _logger = logger;   
+
+            // Create a folder to hold the DPS client certificates. If a folder by the same name already exists, it will be used.
+            s_dpsClientCertificateFolder = Directory.CreateDirectory("DpsClientCertificates");
+        }
+
+        public async Task RunSampleAsync()
+        {
+            try
+            {
+                // Generate the certificate signing request for requesting X509 onboarding certificate for authenticating with IoT hub.
+                // This sample uses openssl to generate an ECC P-256 keypair and the corresponding certificate signing request.
```

Comment:
```suggestion
                // This sample uses openssl to generate an ECC P-256 key/value pair and the corresponding certificate signing request.
```

?

Reply:
This represents the EC public-private key-pair. I'll make it clearer in the comments.